### ISSUE TITLE
Bash/cron command to auto  refresh all podcast feeds

### ISSUE THREAD

**Issue by plum** [tramhao/termusic#304]
## Describe what you want
Is it possible to issue a bash command, (for example in crontab) in order to automaticaaly refresh podcast feeds?
- I.E. Using an existing command to access the termusic database; it seems not , according to the command termusic-server --help

Thanks,
John

**Comment by tramhao** [comment]
Currently only support import feeds and export feeds command for termusic. This could be done I think.

**Comment by RG-IL** [comment]
I already added CLI commands on #733 so i think i could add this to pretty easil, @hasezoey should i do it?

**Comment by RG-IL** [comment]
I could also maybe add CLI commands for the new sorting mechanisms

**Comment by hasezoey** [comment]
> I could also maybe add CLI commands for the new sorting mechanisms

I dont think that is necessary, though as `shuffle` already exists, it would make sense to also have that option.
Though personally i would like to move both options to a `playlist` subcommand, like `termusic playlist shuffle` & `termusic playlist sort MODE`.
(also option `cycle-loop` should also be moved to that `playlist` subcommand)

> I already added CLI commands on https://github.com/tramhao/termusic/pull/733 so i think i could add this to pretty easil, @hasezoey should i do it?

Feel free to do so.
Personally, i would recommend to use a subcommand like `termusic podcast refresh` over `termusic podcast-refresh` so that adding further podcast related options dont end up top-level.
If we do use a new subcommand (`podcast`), then the existing podcast `import` and `export` should be moved there too.

---

Note that the discussion in #720 may be relevant for this. Though that does not apply for just implementing a CLI refresh trigger
TL;DR: the podcast system should be extended to store "last refreshed at" *per podcast* and potentially a "next refresh at" and maybe a a custom time to refresh a podcast (eg one podcast only updates once a month, so once a week is fine, while others update daily).

**Comment by RG-IL** [comment]
> Note that the discussion in [#720](https://github.com/tramhao/termusic/pull/720) may be relevant for this. Though that does not apply for just implementing a CLI refresh trigger TL;DR: the podcast system should be extended to store "last refreshed at" _per podcast_ and potentially a "next refresh at" and maybe a a custom time to refresh a podcast (eg one podcast only updates once a month, so once a week is fine, while others update daily).

I don't fully follow #720 — it looks like a much larger effort involving DB schema changes and auto-refresh scheduling. I don't have the bandwidth to tackle that right now. Should I just focus on the CLI podcast refresh command for the PR?

**Comment by hasezoey** [comment]
> I don't fully follow https://github.com/tramhao/termusic/pull/720 — it looks like a much larger effort involving DB schema changes and auto-refresh scheduling.

It was, just referenced as it might be relevant to this issue overall.

> Should I just focus on the CLI podcast refresh command for the PR?

Sure.

**Comment by RG-IL** [comment]
Ok thanks, i will just make a new PR with the new CLI commands and their reorganization

**Comment by hasezoey** [comment]
This is now possible via `termusic(-server) podcast refresh`, thanks to #750.
